File: ITapply/ITapply.Models/SearchObjects/CandidateSearchObject.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using static ITapply.Models.Responses.EnumResponse;

namespace ITapply.Models.SearchObjects
{
    public class CandidateSearchObject : BaseSearchObject
    {
        public string FirstName { get; set; } = string.Empty;
        public string LastName { get; set; } = string.Empty;
        public string Title { get; set; } = string.Empty;
        public int? LocationId { get; set; }
        public int? MinExperienceYears { get; set; }
        public int? MaxExperienceYears { get; set; }
        public ExperienceLevel? ExperienceLevel { get; set; }
        public string Email { get; set; } = string.Empty;
        public bool? IsActive { get; set; }
    }
} 

File: ITapply/ITapply.UI/itapply_desktop/lib/models/search_objects/candidate_search_object.dart
import 'package:itapply_desktop/models/enums.dart';
import 'package:itapply_desktop/models/search_objects/base_search_object.dart';
import 'package:json_annotation/json_annotation.dart';

part 'candidate_search_object.g.dart';

@JsonSerializable()
class CandidateSearchObject extends BaseSearchObject {
  String? FirstName;
  String? LastName;
  String? Title;
  int? LocationId;
  int? MinExperienceYears;
  int? MaxExperienceYears;
  @JsonKey(name: 'ExperienceLevel', toJson: _experienceLevelToJson)
  ExperienceLevel? experienceLevel;
  String? Email;
  bool? IsActive;

  CandidateSearchObject({
    this.FirstName,
    this.LastName,
    this.Title,
    this.LocationId,
    this.MinExperienceYears,
    this.MaxExperienceYears,
    this.experienceLevel,
    this.Email,
    this.IsActive,
    super.Page,
    super.PageSize,
    super.IncludeTotalCount,
    super.RetrieveAll,
  });

  @override
  Map<String, dynamic> toJson() => _$CandidateSearchObjectToJson(this);
}

int? _experienceLevelToJson(ExperienceLevel? level) => level?.index;